### PR TITLE
Add ocamlfuse.2.7.1-cvs8

### PR DESCRIPTION
**Pull request by astrada**

### CODE REVIEW

**Comment by mseri** [comment]
On the repo it is marked as pre-release: https://github.com/astrada/ocamlfuse/releases
Is that expected?


Looks like it is broken on macos/arm64:
```
#=== ERROR while compiling ocamlfuse.2.7.1-cvs8 ===============================#
# context              2.1.4 | macos/arm64 | ocaml-base-compiler.5.0.0 | pinned(https://github.com/astrada/ocamlfuse/archive/v2.7.1_cvs8.tar.gz)
# path                 ~/.opam/5.0.0/.opam-switch/build/ocamlfuse.2.7.1-cvs8
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p ocamlfuse -j 7 @install
# exit-code            1
# env-file             ~/.opam/log/ocamlfuse-86053-63ce51.env
# output-file          ~/.opam/log/ocamlfuse-86053-63ce51.out
### output ###
# (cd _build/default/lib && /usr/bin/cc -O2 -fno-strict-aliasing -fwrapv -pthread -D_FILE_OFFSET_BITS=64 -D_FILE_OFFSET_BITS=64 -I/usr/local/include/fuse -g -I /Users/mac1000/.opam/5.0.0/lib/ocaml -I /Users/mac1000/.opam/5.0.0/lib/camlidl -I /Users/mac1000/.opam/5.0.0/lib/ocaml/threads -I /Users/mac1000/.opam/5.0.0/lib/ocaml/unix -o Unix_util_stubs.o -c Unix_util_stubs.c)
# Unix_util_stubs.c:151:3: warning: expression result unused [-Wunused-value]
#   bufv;
#   ^~~~
# 1 warning generated.
# File "lib/dune", line 8, characters 9-28:
# 8 |   (names Fuse_bindings_stubs Fuse_util Unix_util_stubs)
#              ^^^^^^^^^^^^^^^^^^^
# (cd _build/default/lib && /usr/bin/cc -O2 -fno-strict-aliasing -fwrapv -pthread -D_FILE_OFFSET_BITS=64 -D_FILE_OFFSET_BITS=64 -I/usr/local/include/fuse -g -I /Users/mac1000/.opam/5.0.0/lib/ocaml -I /Users/mac1000/.opam/5.0.0/lib/camlidl -I /Users/mac1000/.opam/5.0.0/lib/ocaml/threads -I /Users/mac1000/.opam/5.0.0/lib/ocaml/unix -o Fuse_bindings_stubs.o -c Fuse_bindings_stubs.c)
# Fuse_bindings_stubs.c:698:3: error: implicit declaration of function 'enter_blocking_section' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
#   enter_blocking_section();
#   ^
# Fuse_bindings_stubs.c:698:3: note: did you mean 'caml_enter_blocking_section'?
# /Users/mac1000/.opam/5.0.0/lib/ocaml/caml/signals.h:30:17: note: 'caml_enter_blocking_section' declared here
# CAMLextern void caml_enter_blocking_section (void);
#                 ^
# Fuse_bindings_stubs.c:700:3: error: implicit declaration of function 'leave_blocking_section' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
#   leave_blocking_section();
#   ^
# Fuse_bindings_stubs.c:700:3: note: did you mean 'enter_blocking_section'?
# Fuse_bindings_stubs.c:698:3: note: 'enter_blocking_section' declared here
#   enter_blocking_section();
#   ^
# Fuse_bindings_stubs.c:714:3: error: implicit declaration of function 'enter_blocking_section' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
#   enter_blocking_section();
#   ^
# Fuse_bindings_stubs.c:716:3: error: implicit declaration of function 'leave_blocking_section' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
#   leave_blocking_section();
#   ^
# Fuse_bindings_stubs.c:759:3: error: implicit declaration of function 'enter_blocking_section' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
#   enter_blocking_section();
#   ^
# Fuse_bindings_stubs.c:761:3: error: implicit declaration of function 'leave_blocking_section' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
#   leave_blocking_section();
#   ^
# 6 errors generated.
# (cd _build/default/lib && /usr/bin/cc -O2 -fno-strict-aliasing -fwrapv -pthread -D_FILE_OFFSET_BITS=64 -D_FILE_OFFSET_BITS=64 -I/usr/local/include/fuse -g -I /Users/mac1000/.opam/5.0.0/lib/ocaml -I /Users/mac1000/.opam/5.0.0/lib/camlidl -I /Users/mac1000/.opam/5.0.0/lib/ocaml/threads -I /Users/mac1000/.opam/5.0.0/lib/ocaml/unix -o Fuse_util.o -c Fuse_util.c)
# Fuse_util.c:632:13: warning: add explicit braces to avoid dangling else [-Wdangling-else]
# FOR_ALL_OPS(CALLBACK)
#             ^
# Fuse_util.c:644:15: warning: incompatible function pointer types assigning to 'int (*)(const char *, const char *, const char *, size_t, int, uint32_t)' (aka 'int (*)(const char *, const char *, const char *, unsigned long, int, unsigned int)') from 'int (const char *, const char *, const char *, size_t, int)' (aka 'int (const char *, const char *, const char *, unsigned long, int)') [-Wincompatible-function-pointer-types]
#   FOR_ALL_OPS(SET_OPERATION)
#   ~~~~~~~~~~~~^~~~~~~~~~~~~~
# Fuse_util.c:328:3: note: expanded from macro 'FOR_ALL_OPS'
#   MACRO(setxattr)                               \
#   ^~~~~~~~~~~~~~~
# Fuse_util.c:639:15: note: expanded from macro 'SET_OPERATION'
#     ops.OPNAME=ops_##OPNAME;                        \
#               ^~~~~~~~~~~~~
# Fuse_util.c:644:15: warning: incompatible function pointer types assigning to 'int (*)(const char *, const char *, char *, size_t, uint32_t)' (aka 'int (*)(const char *, const char *, char *, unsigned long, unsigned int)') from 'int (const char *, const char *, char *, size_t)' (aka 'int (const char *, const char *, char *, unsigned long)') [-Wincompatible-function-pointer-types]
#   FOR_ALL_OPS(SET_OPERATION)
#   ~~~~~~~~~~~~^~~~~~~~~~~~~~
# Fuse_util.c:329:3: note: expanded from macro 'FOR_ALL_OPS'
#   MACRO(getxattr)                               \
#   ^~~~~~~~~~~~~~~
# Fuse_util.c:639:15: note: expanded from macro 'SET_OPERATION'
#     ops.OPNAME=ops_##OPNAME;                        \
#               ^~~~~~~~~~~~~
# 3 warnings generated.
```

File: packages/ocamlfuse/ocamlfuse.2.7.1-cvs8/opam [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+authors: [
+  "Vincenzo Ciancia"
+  "Olaf Hering <olaf@aepfle.de>"
+  "Alessandro Strada <alessandro.strada@gmail.com>"
+]
+license: "GPL-1.0-or-later"
+bug-reports: "https://github.com/astrada/ocamlfuse/issues"
+depends: [
+  "dune" {>= "3.7"}
+  "ocaml" {>= "4.02.3"}
```
Comment (kit-ty-kate) [inline packages/ocamlfuse/ocamlfuse.2.7.1-cvs8/opam:21]: See failure above
```suggestion
  "ocaml" {>= "4.02.3" & < "5.0"}
```

Comment (astrada) [reply packages/ocamlfuse/ocamlfuse.2.7.1-cvs8/opam:21]: The problem seems to be caused by camlidl. It should be solved by this PR: https://github.com/xavierleroy/camlidl/pull/25. When it's merged, it should fix the error on macos with ocaml 5.0. The purpose of this new version is indeed supporting ocaml 5.0.

**Comment by astrada** [comment]
> On the repo it is marked as pre-release: https://github.com/astrada/ocamlfuse/releases

Fixed. I usually set it as pre-release until I'm sure I don't have to release a hotfix.

**Comment by avsm** [comment]
Just waiting for a camlidl release, so marking this as temporarily blocked. Unless you'd like to mark it as `!macos` and get it in the repo for Linux, @astrada ?

**Comment by astrada** [comment]
> Just waiting for a camlidl release, so marking this as temporarily blocked. Unless you'd like to mark it as `!macos` and get it in the repo for Linux, @astrada ?

OK, I pushed an opam file that disable macos

File: packages/ocamlfuse/ocamlfuse.2.7.1-cvs8/opam [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
+    "@doc" {with-doc}
+  ]
+]
+available: [os != "macos"]
```
Comment (mseri) [inline packages/ocamlfuse/ocamlfuse.2.7.1-cvs8/opam:44]:
```suggestion
```
In this case I suggest you let camlidl fail (I will fix camlidl opam file separately) and when a new version is released, if the CI fails, I will manually mark this as unavailable on macos

**Pull request merged**